torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1949,6 +1949,29 @@ void ONNXShapeTypeInference(
   SetGraphInputTypeReliable(n->owningGraph());
   GRAPH_UPDATE(
       "Running ONNX shape inference for node: ", n->kind().toDisplayString());
+
+  // Go through every output to check if they all have shape
+  // If they all do, this should be reliable even if the Op is not from ONNX.
+  size_t all_output_has_value = 0;
+  for (auto node_output : n->outputs()) {
+    if (auto output_type = node_output->type()->cast<TensorType>()) {
+      if (output_type->dim().has_value()) {
+        // Go through every symbolic_sizes and if one of them > 0, we say this
+        // is set by user. On the other hand, if all of them are *, we take this
+        // node does not have given type, since unreliable nodes have * shape
+        // anyway.
+        for (size_t i = 0; i < output_type->dim().value(); i++) {
+          if (output_type->symbolic_sizes()[i].value() > 0) {
```

Comment:
nit: wonder if can simplify the loop to just check if has `symbolic_sizes`, and if so loop over it.

Reply:
Found and used is_static() and symbolic_sizes().sizes(), and they come in handy.